### ISSUE TITLE
--source-path doesn't work at all

### ISSUE THREAD

**Issue by YorkZ** [koalaman/shellcheck#1998]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1090
- My shellcheck version (`shellcheck --version` or "online"): 0.7.1

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

this_file=$(readlink -f "${BASH_SOURCE[0]}")
this_dir=$(dirname "${this_file}")
# shellcheck source-path=SCRIPTDIR
source "${this_dir}/menu-common.sh"
```
#### Here's my .shellcheckrc (in both my home directory and the script directory:
```
source-path=SCRIPTDIR
```

### I also tried passing `source-path` on the command line:
```sh
shellcheck --source-path=SCRIPTDIR customer-menu
```

#### Here's what shellcheck currently says:
Can't follow non-constant source. Use a directive to specify location. [SC1090]

#### Here's what I wanted or expected to see:
No warning

**Comment by koalaman** [comment]
I see why this is confusing: the wiki had been incorrectly edited to say that this would work. The intended behavior of `source-path` was just to root constant relative or absolute paths like `./foo` or `/usr/bin/foo`.

It's been discussed a bit before, and now ShellCheck will strip a leading expansion followed by a slash and treat it as `./`. This allows the most common idioms like `source "${this_dir}/file.sh` to work without explicit `source=file.sh` type directives.

Paths with overrides, constant paths, or paths with multiple expansions will be treated like before. 

Thanks!

**Comment by jonhoo** [comment]
fwiw, I found that what I needed was:
```bash
# shellcheck source-path=SCRIPTDIR source=name-of-script
```

**Comment by 0x656b694d** [comment]
According to the docs `source-path=SCRIPTDIR` should be enough, however I haven't found a way to make this pass:
```
# shellcheck source-path=SCRIPTDIR
source "$SCRIPTS_ROOT/common.sh"
```
(The script with these lines and `common.sh` are in the same directory.)

It still reports `SC1091: Not following: ./common.sh was not specified as input (see shellcheck -x)`

**Comment by speedythesnail** [comment]
> According to the docs `source-path=SCRIPTDIR` should be enough, however I haven't found a way to make this pass:
> 
> ```
> # shellcheck source-path=SCRIPTDIR
> source "$SCRIPTS_ROOT/common.sh"
> ```
> 
> (The script with these lines and `common.sh` are in the same directory.)
> 
> It still reports `SC1091: Not following: ./common.sh was not specified as input (see shellcheck -x)`

I would like to second this. I have the following:
```
#!/bin/bash
#######################################################################################################
# Get current script dir and other dirs
#######################################################################################################
# Get script directory, even with symbolic links
# https://stackoverflow.com/a/246128/18121299
SOURCE=${BASH_SOURCE[0]}
while [ -L "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
  SCRIPT_DIR=$( cd -P "$( dirname "$SOURCE" )" >/dev/null 2>&1 && pwd )
  SOURCE=$(readlink "$SOURCE")
  [[ $SOURCE != /* ]] && SOURCE=$SCRIPT_DIR/$SOURCE # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
done
SCRIPT_DIR=$( cd -P "$( dirname "$SOURCE" )" >/dev/null 2>&1 && pwd )

# Ensure we are in the script dir
pushd "$SCRIPT_DIR" > /dev/null || exit
# Project is one level up
pushd .. > /dev/null

PROJECT_DIR="$(pwd)"
JSON_DIR="${PROJECT_DIR}/json"

# Return to the starting dir
popd > /dev/null || exit
popd > /dev/null || exit

export SCRIPT_DIR
export PROJECT_DIR
export JSON_DIR

# shellcheck source-path=SCRIPTDIR
source "$SCRIPT_DIR/record_types.sh"
```

I too am receiving the "Not following: ./record_types.sh was not specified as input (see shellcheck -x).shellcheck[SC1091](https://www.shellcheck.net/wiki/SC1091)" message.

**Comment by koalaman** [comment]
Run `shellcheck -x` or specify `external-sources=true` in `.shellcheckrc`.

ShellCheck started life as an online tool that would run on random strangers' code in a time before containerization, so it doesn't give itself permission to read arbitrary files around the filesystem just because an untrusted script asked them to. If it had been designed today as the local tool it has become, it would have had no problems doing that.
